[jit] add `inlined_graph` method to ScriptFunctions

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33515 [jit] Fix ModuleDict type sharing
* **#33508 [jit] add `inlined_graph` method to ScriptFunctions**

Ever since we switched to not inlining by default, some users have
complained since they relied on inlining occuring to, e.g. process the
graph with some other tool. Add an inlined_graph for convenience in
those cases.

Differential Revision: [D19977638](https://our.internmc.facebook.com/intern/diff/D19977638)

suo committed:
[jit] add `inlined_graph` method to ScriptFunctions

Ever since we switched to not inlining by default, some users have
complained since they relied on inlining occuring to, e.g. process the
graph with some other tool. Add an inlined_graph for convenience in
those cases.

[ghstack-poisoned]

suo committed:
Update on "[jit] add `inlined_graph` method to ScriptFunctions"


Ever since we switched to not inlining by default, some users have
complained since they relied on inlining occuring to, e.g. process the
graph with some other tool. Add an inlined_graph for convenience in
those cases.

[ghstack-poisoned]

driazati reviewed (approved):
lgtm minus the sub module change, can you also add a `FileCheck` test to make sure it works

suo committed:
Update on "[jit] add `inlined_graph` method to ScriptFunctions"


Ever since we switched to not inlining by default, some users have
complained since they relied on inlining occuring to, e.g. process the
graph with some other tool. Add an inlined_graph for convenience in
those cases.

[ghstack-poisoned]

suo committed:
Update on "[jit] add `inlined_graph` method to ScriptFunctions"


Ever since we switched to not inlining by default, some users have
complained since they relied on inlining occuring to, e.g. process the
graph with some other tool. Add an inlined_graph for convenience in
those cases.

[ghstack-poisoned]
